File: pset6/readability/readability.py
# create variable to hold text, leght of text, letter count, word count, and sentences respectively
the_text = input("Text: ")
length_of_text = len(the_text)
letters = 0
words = 1
sentences = 0

# go through the input text character by character
for i in range(length_of_text):
    c = the_text[i]

    # if c is alphabetical then add 1 to letters
    if c.isalpha():
        letters += 1
    # if c is a space then add 1 to words
    if c.isspace():
        words += 1
    # if c is a '.' or '!' or '?' add 1 to sentences
    if c == '.' or c == '!' or c == '?':
        sentences += 1

# calculate Coleman-Liau index and round off. set grade to result
grade = round(0.0588 * (letters * 100.0 / words) - 0.296 * (sentences * 100.0 / words) - 15.8)

# print out the greade level
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

int main(void)
{
    int letters = 0, words = 1, sentences = 0, grade;
    char c;
    // get a string from the user
    string the_text = get_string("Text: ");
    for (int i = 0, n = strlen(the_text); i < n; i++)
    {
        c = the_text[i];
        // if c is a letter add one to letter count
        if (isalpha(c))
        {
            letters++;
        }
        // if c is a space add one to words count
        if (isspace(c))
        {
            words++;
        }
        // if the sentence ends with . or ! or ? add one to sentences count
        if (c == '.' || c == '!'  || c == '?')
        {
            sentences++;
        }
    }

    // get average grade using: index = 0.0588 * L - 0.296 * S - 15.8
    // where L is the average number of letters per 100 words in the
    // text, and S is the average number of sentences per 100 words
    // in the text.
    grade = round(0.0588 * (letters * 100.0 / words) - 0.296 * (sentences * 100.0 / words) - 15.8);

    // print out grade level
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}